FreeRTOS-Plus/Demo/Device_Shadow_Windows_Simulator/Device_Shadow_Demo/DemoTasks/ShadowDemoMainExample.c

```diff
@@ -0,0 +1,756 @@
+                          SHADOW_REPORTED_JSON,
+                          ( int ) ulCurrentPowerOnState,
+                          ( long unsigned ) ulClientToken );
+
+                returnStatus = lPublishToTopic( SHADOW_TOPIC_STRING_UPDATE( THING_NAME ),
+                                               SHADOW_TOPIC_LENGTH_UPDATE( THING_NAME_LENGTH ),
+                                               pcUpdateDocument,
+                                               ( SHADOW_DESIRED_JSON_LENGTH + 1 ) );
+            }
+            else
+            {
+                LogInfo( ( "No change from /update/delta, unsubscribe all shadow topics and disconnect from MQTT.\r\n" ) );
+            }
+        }
+
+        if( returnStatus == EXIT_SUCCESS )
+        {
+            LogInfo( ( "Start to unsubscribe shadow topics and disconnect from MQTT. \r\n" ) );
+
+            returnStatus = lUnsubscribeFromTopic( SHADOW_TOPIC_STRING_UPDATE_DELTA( THING_NAME ),
+                                                 SHADOW_TOPIC_LENGTH_UPDATE_DELTA( THING_NAME_LENGTH ) );
+
+            if( returnStatus != EXIT_SUCCESS )
+            {
+                LogError( ( "Failed to unsubscribe the topic %s",
+                            SHADOW_TOPIC_STRING_UPDATE_DELTA( THING_NAME ) ) );
+            }
+        }
+
+        if( returnStatus == EXIT_SUCCESS )
+        {
+            returnStatus = lUnsubscribeFromTopic( SHADOW_TOPIC_STRING_UPDATE_ACCEPTED( THING_NAME ),
+                                                 SHADOW_TOPIC_LENGTH_UPDATE_ACCEPTED( THING_NAME_LENGTH ) );
+
+            if( returnStatus != EXIT_SUCCESS )
+            {
+                LogError( ( "Failed to unsubscribe the topic %s",
+                            SHADOW_TOPIC_STRING_UPDATE_ACCEPTED( THING_NAME ) ) );
+            }
+        }
+
+        if( returnStatus == EXIT_SUCCESS )
+        {
+            returnStatus = lUnsubscribeFromTopic( SHADOW_TOPIC_STRING_UPDATE_REJECTED( THING_NAME ),
+                                                 SHADOW_TOPIC_LENGTH_UPDATE_REJECTED( THING_NAME_LENGTH ) );
+
+            if( returnStatus != EXIT_SUCCESS )
+            {
+                LogError( ( "Failed to unsubscribe the topic %s",
+                            SHADOW_TOPIC_STRING_UPDATE_REJECTED( THING_NAME ) ) );
+            }
+        }
+
+        /* The MQTT session is always disconnected, even there were prior failures. */
+        returnStatus = lDisconnectMqttSession();
+
+        /* This demo performs only Device Shadow operations. If matching the Shadow
+         * MQTT topic fails or there are failure in parsing the received JSON document,
+         * then this demo was not successful. */
+        if( ( lUpdateAcceptedReturn != EXIT_SUCCESS ) || ( lUpdateDeltaReturn != EXIT_SUCCESS ) )
+        {
+            LogError( ( "Callback function failed." ) );
+        }
+
+        if( returnStatus == EXIT_SUCCESS )
+        {
+            LogInfo( ( "Shadow Demo is functional." ) );
```

Comment:
Please change to "Demo completed successfully." It is how CI reports success or failure.